src/NServiceBus.Core/DelayedDelivery/TimeoutManager/TimeoutRecoverabilityBehavior.cs

```diff
@@ -0,0 +1,115 @@
+
+            if (ShouldAttemptAnotherRetry(failureInfo))
+            {
+                try
+                {
+                    await next().ConfigureAwait(false);
+                    return;
+                }
+                catch (Exception exception)
+                {
+                    HandleProcessingFailure(context, message, exception, failureInfo);
+
+                    context.AbortReceiveOperation();
+                    return;
+                }
+            }
+
+            GiveUpForMessage(message, failureInfo);
+
+            await MoveToErrorQueue(context, message, failureInfo);
+        }
+
+        bool ShouldAttemptAnotherRetry(ProcessingFailureInfo failureInfo)
+        {
+            if (failureInfo == null)
+            {
+                return true;
+            }
+            return failureInfo.NumberOfFailedAttempts <= MaxNumberOfFailedRetries;
+        }
+
+        void HandleProcessingFailure(ITransportReceiveContext context, IncomingMessage message, Exception exception, ProcessingFailureInfo failureInfo)
+        {
+            failures.RecordFailureInfoForMessage(message.MessageId, exception);
+
+            Logger.Info($"First Level Retry is going to retry message '{message.MessageId}' because of an exception:", exception);
+
+            var numberOfFailedAttempts = failureInfo?.NumberOfFailedAttempts ?? 0;
+            notifications.Errors.InvokeMessageHasFailedAFirstLevelRetryAttempt(numberOfFailedAttempts, context.Message, exception);
+        }
+
+        void GiveUpForMessage(IncomingMessage message, ProcessingFailureInfo failureInfo)
+        {
+            message.Headers[Headers.FLRetries] = failureInfo.NumberOfFailedAttempts.ToString();
+
+            notifications.Errors.InvokeMessageHasFailedAFirstLevelRetryAttempt(failureInfo.NumberOfFailedAttempts, message, failureInfo.Exception);
+
+            failures.ClearFailureInfoForMessage(message.MessageId);
+
+            Logger.Info($"Giving up First Level Retries for message '{message.MessageId}'.");
+        }
+
+        async Task MoveToErrorQueue(ITransportReceiveContext context, IncomingMessage message, ProcessingFailureInfo failureInfo)
+        {
+            try
+            {
+                Logger.Error($"Moving timeout message '{message.MessageId}' to the error queue because processing failed due to an exception:", failureInfo.Exception);
+
+                message.SetExceptionHeaders(failureInfo.Exception, localAddress);
+
+                var outgoingMessage = new OutgoingMessage(message.MessageId, message.Headers, message.Body);
+                var routingStrategy = new UnicastRoutingStrategy(errorQueueAddress);
+                var addressTag = routingStrategy.Apply(new Dictionary<string, string>());
+
+                var transportOperations = new TransportOperations(new TransportOperation(outgoingMessage, addressTag, DispatchConsistency.Default));
```

Comment:
Can we remove the default value from the invocation? Probably does not give much data.